advanced/src/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { HttpLink, InMemoryCache, ApolloClient } from 'apollo-client-preset'
+import { WebSocketLink } from 'apollo-link-ws'
+import { ApolloLink, split } from 'apollo-link'
+import { getMainDefinition } from 'apollo-utilities'
+import { AUTH_TOKEN } from './constant'
+import SuperContainer from './components/SuperContainer'
+
+import 'tachyons'
+import './index.css'
+
+const httpLink = new HttpLink({ uri: 'http://localhost:4000' })
+
+const middlewareLink = new ApolloLink((operation, forward) => {
+  // get the authentication token from local storage if it exists
+  const tokenValue = localStorage.getItem(AUTH_TOKEN)
+  // return the headers to the context so httpLink can read them
+  operation.setContext({
+    headers: {
+      Authorization: tokenValue ? `Bearer ${tokenValue}` : "",
+    }
+  })
+  return forward(operation)
+})
+
+// Authenticated httplink
+const httpLinkAuth = middlewareLink.concat(httpLink)
+
+const wsLink = new WebSocketLink({
+  uri: `wss://localhost:4000`,
+  options: {
+    reconnect: true,
+    connectionParams: {
+      Authorization: `Bearer ${localStorage.getItem(AUTH_TOKEN)}`
+    }
+  }
+})
+
+const link = split(
+  // split based on operation type
+  ({ query }) => {
+    const { kind, operation } = getMainDefinition(query)
+    return kind === 'OperationDefinition' && operation === 'subscription'
+  },
+  wsLink,
+  httpLinkAuth,
+)
+
+// apollo client setup
+const client = new ApolloClient({
+  link: ApolloLink.from([link]),
+  cache: new InMemoryCache(),
+  connectToDevTools: true
+})
+
+const token = localStorage.getItem (AUTH_TOKEN)
+
+function RenderDom() {
+  ReactDOM.render(<SuperContainer client={client} token={token}/> ,document.getElementById('root'))
+}
+
+RenderDom()
```

Comment:
Was there a reason for putting the `render` method in a function and calling that function afterwards?

Reply:
No reason I suppose I first had the SuperContainer here which is the reason i did that.
Removed already :)